rayon-core/src/lib.rs

```diff
@@ -84,9 +85,21 @@ pub fn current_num_threads() -> usize {
     ::registry::Registry::current_num_threads()
 }
 
+/// Error when initializing a thread pool
+#[derive(Debug)]
+pub struct ThreadPoolInitError {
+    kind: ErrorKind,
+}
+
+#[derive(Debug)]
+enum ErrorKind {
+    GlobalPoolAlreadyInitialized,
+    IOError(io::Error),
+}
+
 /// Contains the rayon thread pool configuration.
```

Comment:
I think we could do a bit better here in this documentation. =) Maybe something like this:

---

Used to configure and create a new thread-pool, or to configure the global rayon thread-pool. 

# Examples

The following creates a Rayon thread-pool using 22 threads:

```rust
let pool = ThreadPoolBuilder::new().num_threads(22).build();
```

You can also configure the global thread pool by using `init_global` instead of `build`:

```rust
ThreadPoolBuilder::new().num_threads(22).init_global();
```